File: guides/micronaut-mongodb-asynchronous/java/src/main/java/example/micronaut/FruitRepository.java
package example.micronaut;

import io.micronaut.core.annotation.NonNull;
import org.reactivestreams.Publisher;
import reactor.core.publisher.Mono;

import javax.validation.Valid;
import javax.validation.constraints.NotNull;

public interface FruitRepository {

    @NonNull
    Publisher<Fruit> list();

    Mono<Boolean> save(@NonNull @NotNull @Valid Fruit fruit); // <1>
}

File: guides/micronaut-eclipsestore-persitence/kotlin/src/main/kotlin/example/micronaut/FruitRepository.kt
package example.micronaut

import jakarta.validation.Valid

interface FruitRepository {

    fun list(): Collection<Fruit>

    fun create(@Valid fruit: FruitCommand): Fruit // <1>

    fun update(@Valid fruit: FruitCommand): Fruit? // <1>

    fun find(name: String): Fruit?

    fun delete(@Valid fruit: FruitCommand) // <1>
}